content/Link Tracking with StimulusReflex - Part III.md

```diff
@@ -0,0 +1,159 @@
+# Link Tracking with StimulusReflex - Part III
+
+In this last part of this series, we'll use what [we built before](TODO: Add link to second part) to actually start tracking views. To do so, we'll use [ActionCable](https://guides.rubyonrails.org/action_cable_overview.html) and [CableReady](https://github.com/hopsoft/cable_ready) to broadcast and make changes in our frontend.
+
+![Header image](images/stimulus_reflex.jpeg)
+
+### Redirecting users
+
+Let's start with the simple stuff, when a user follows one of our shortened links, they expect to be redirected to the original URL. To do this, we'll create a new controller and add the corresponding route.
+
+```ruby
+# app/controllers/redirections_controller.rb
+
+class RedirectionsController < ApplicationController
+  def index
+  end
+end
+```
+
+```ruby
+# config/routes.rb
+
+Rails.application.routes.draw do
+  # ...
+  get '/:path', to: 'redirections#index'
+end
+```
+
+Remember to add it at the bottom of the block so it does not override any other routes.
+
+With this, the implementation should be pretty straightforward:
+
+```ruby
+# app/controllers/redirections_controller.rb
+
+class RedirectionsController < ApplicationController
+  def index
+    link = ShortenedLink.find_by!(shortened_path: params[:path])
+    link.increment!(:views_count)
+
+    redirect_to link.original_url
+  end
+end
+```
+
+We want to use [`#increment!`](https://apidock.com/rails/ActiveRecord/Base/increment!) since it's atomic and we are sure we don't want to run any validations, which would just slow us down.
```

Comment:
```suggestion
We want to use [`#increment!`](https://apidock.com/rails/ActiveRecord/Base/increment!) since it's atomic, and we are sure we don't want to run any validations, which would just slow us down.
```